services/project_init_service.py

```
@@ -0,0 +1,216 @@
+            config_file.write_text(config_content, encoding="utf-8")
+        else:
+            # 用户选择 yaml 时，删除默认的 hugo.toml 并写入 hugo.yaml
+            default_toml = site_root / "hugo.toml"
+            if default_toml.exists() and config_format == "yaml":
+                default_toml.unlink()
+            config_file.write_text(config_content, encoding="utf-8")
+
+    @staticmethod
+    def switch_active_project(app, registry, new_root: Path | str) -> None:
+        """
+        将活跃 Hugo 项目切换到新目录，并重新初始化所有依赖服务。
```

Ruff (0.15.18)

[warning] Docstring contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF002)